src/index/base.cpp

```diff
@@ -14,6 +14,8 @@
 #include <validation.h> // For g_chainman
 #include <warnings.h>
 
+using node::ReadBlockFromDisk;
```

Comment:
What is the rule here? Randomly pick `using` or specify the namespace before the symbol?

Reply:
> What is the rule here? Randomly pick `using` or specify the namespace before the symbol?

I don't know if there's a rule, but I think `using some_namespace::SomeSymbol;` statements are generally a bad thing, and should be used sparingly. They are bad because they can make it unclear where symbols are coming from, and can lead to ugly code when longer names and shorter names are used inconsistently.

In this PR, I'm trying to limit scope of `using` statements and only adding them locally in .cpp files, never in header files. I think most of the places where they are added are places where code organization should be improved, and that these will be removed when it is improved. For example, I'm working on porting src/index/ code to use `interfaces::Chain` and stop using node code and `node::` symbols. Carl and James and others are moving more validation code to `src/node/` which will make `using` statements in validation code go away. The `using` statements in RPC, bench, and test code could potentially be removed by having a `src/node/test/`directory analogous to the existing `src/wallet/test/` directory and a `src/node/rpc/` directory analagous to the existing `src/wallet/rpc/` directory. Or they could just be removed by adding `node::` prefixes to relevant types and symbols. I think either of these changes would be a win for clarity and organization, and the goal of this PR is to enable that.

Reply:
> Carl and James and others are moving more validation code to `src/node/` which will make `using` statements in validation code go away.

Actually, I'm assuming too much here. Validation code doesn't have to move in order to avoid `using node::` statements  or `node::` prefixes. It could alternately become more modular and stop depending directly on `node::` types and symbols. It could move someplace other than src/node/, like src/kernel/, or not move at all. The number of using statements and node:: references goes down in any case.

Reply:
Agree that the `using` statements should be used sparingly.

If I understand correctly, the hope is that the `using` statements that you have right now will be removed in the future as we find better interfaces/boundaries between our modules? I can see how that would work for src/index/ code and `interfaces::Chain`, but I'm wondering how we'd get rid of cases like the various places with `using node::NodeContext;`, or perhaps these cases are ones where we want to keep the `using` statements?

Reply:
> If I understand correctly, the hope is that the `using` statements that you have right now will be removed in the future as we find better interfaces/boundaries between our modules?

Yes, but I think this change is good whether or not that happens. The `using` statements just show places where module boundaries are being crossed. In general it's better if modules are self-contained and boundaries are crossed in fewer places. But it's useful in any case to be able to see the places where boundaries are crossed and not have hidden dependencies.

> I can see how that would work for src/index/ code and `interfaces::Chain`, but I'm wondering how we'd get rid of cases like the various places with `using node::NodeContext;`, or perhaps these cases are ones where we want to keep the `using` statements?

Just if you move the code using `node::NodeContext` *inside* the `node` namespace, then you no longer need `node::` prefixes or `using` statements to reference it. This is what I was referring to with the `src/node/rpc/` `src/node/test/` idea above, but that's just one possible way the node module could be more self-contained.

(Also `NodeContext` is kind of a weird case because it's the one node identifier that begins with word "node". I was thinking of making a followup PR that would remove all `using node::NodeContext` and rename `node::NodeContext` to `node::Context` and rename `wallet::WalletContext` to `wallet::Context`. Would be purely an aesthetic change, though, and low on the priority list.)